editor/plugins/abstract_polygon_2d_editor.cpp

```diff
@@ -437,24 +434,28 @@ bool AbstractPolygon2DEditor::forward_gui_input(const Ref<InputEvent> &p_event)
 
 			canvas_item_editor->update_viewport();
 		} else if (mode == MODE_EDIT || (_is_line() && mode == MODE_CREATE)) {
-			const PosVertex onEdgeVertex = closest_edge_point(gpoint);
-
-			if (onEdgeVertex.valid()) {
-				hover_point = Vertex();
-				edge_point = onEdgeVertex;
+			const PosVertex new_hover_point = closest_point(gpoint);
+			if (hover_point != new_hover_point) {
+				hover_point = new_hover_point;
 				canvas_item_editor->update_viewport();
-			} else {
-				if (edge_point.valid()) {
-					edge_point = PosVertex();
-					canvas_item_editor->update_viewport();
-				}
+			}
 
-				const PosVertex new_hover_point = closest_point(gpoint);
-				if (hover_point != new_hover_point) {
-					hover_point = new_hover_point;
+			bool edge_hover = false;
+			if (!hover_point.valid()) {
+				const PosVertex onEdgeVertex = closest_edge_point(gpoint);
```

Comment:
Do we want to fix the naming of this while were here?

Reply:
Usually I'd say no as it would make the diff needlessly hard to compare, but here it seems to be the only occurrence of camelCase in surrounding code, so might be worth fixing yeah.